File: src/Str.php
<?php

namespace xltxlm\str;

use xltxlm\arr\Arr;

/**
 * 字符串类型的基本操作函数集合;
 */
class Str extends Str\Str_implements
{
    public function Lowercase(): string
    {
        $this->setValue(strtolower($this->Value));
        return $this->getValue();
    }

    /**
     * Str constructor.
     */
    public function __construct(string $Value = '')
    {
        if ($Value) {
            $this->setValue(trim($Value));
        }
    }

    public function getLastChar(): string
    {
        return substr($this->getValue(), -1);
    }


    public function Split(string $delimiter = null): \xltxlm\Arr\Arr
    {
        $explode = explode($delimiter, $this->getValue());
        //再去掉空的字符占位
        $explode = array_diff($explode, [""]);
        $explodeObject = (new Arr())
            ->setValues($explode);
        return $explodeObject;
    }

    public function Strtr(string $oldvar = null, string $newvar = null): string
    {
        $var = strtr($this->getValue(), [$oldvar => $newvar]);
        $this->setValue($var);
        return $var;
    }

    public function Ucfirst(): string
    {
        return ucfirst($this->getValue());
    }

    public function Lcfirst(): string
    {
        return lcfirst($this->getValue());
    }

    public function Substr(int $Index = null, $Length = null): string
    {
        $var = mb_substr($this->getValue(), $Index, $Length ?: strlen($this->getValue()));
        $this->setValue($var);
        return $var;
    }

    public function Strpos(string $Compare_String = null): bool
    {
        return strpos($this->getValue(), $Compare_String) !== false;
    }


    public function One_Str_by_Index(int $Index = null): string
    {
        return $this->Substr($Index, 1);
    }

    public function getJson(): bool
    {
        $first = (new Str($this->getValue()))
                ->Substr(0, 1) == '[';
        $last = (new Str($this->getValue()))
                ->Substr(-1, 1) == ']';

        if ($first && $last) {
            return !empty(json_decode($this->getValue(), true));
        }
        $first = (new Str($this->getValue()))
                ->Substr(0, 1) == '{';
        $last = (new Str($this->getValue()))
                ->Substr(-1, 1) == '}';

        if ($first && $last) {
            return !empty(json_decode($this->getValue(), true));
        }
        return false;
    }

    /**
     *  判断某个字符串是不是原始字符串的结尾;
     *
     * @return :bool;
     */
    public function isEndofstr(string $newstr = null): bool
    {
        $new_strlen = strlen($newstr);
        if ($new_strlen == 0) {
            return false;
        }
        $index = strrpos($this->getValue(), $newstr);
        if ($index + $new_strlen == strlen($this->getValue())) {
            return true;
        }
        return false;
    }


}

File: go/Str.go
package 

type Str struct {
    /* 要处理的字符串内容 */
    Value string

    /* 获取最后一个字符串 */
    LastChar string

}

func NewStr(Value string,LastChar string) *Str{
    var this = new(Str)
    this.SetValue(Value);
    this.SetLastChar(LastChar);
    return this
}

func (this *Str)GetValue() string{
    return this.Value;
}

func (this *Str)SetValue(Value string) *Str{
    this.Value = Value;
    return this
}
func (this *Str)GetLastChar() string{
    return this.LastChar;
}

func (this *Str)SetLastChar(LastChar string) *Str{
    this.LastChar = LastChar;
    return this
}

/**
    */
func (this *Str)__toString()string{

}

/**
    根据指定的字符串,砍成数组对象,注意:会去掉空内容*/
func (this *Str)Split(delimiter string){

}

/**
    字符串替换*/
func (this *Str)Strtr(newvar string,oldvar string)string{

}
func (this *Str)Strtr_this(newvar string,oldvar string) *Str{
    this.Strtr(newvar,oldvar);
    return this;
}

/**
    首字母大写*/
func (this *Str)Ucfirst()string{

}
func (this *Str)Ucfirst_this() *Str{
    this.Ucfirst();
    return this;
}

/**
    首字母小写*/
func (this *Str)Lcfirst()string{

}
func (this *Str)Lcfirst_this() *Str{
    this.Lcfirst();
    return this;
}

/**
    从指定索引开始砍字符串*/
func (this *Str)Substr(Index int,Length )string{

}
func (this *Str)Substr_this(Index int,Length ) *Str{
    this.Substr(Index,Length);
    return this;
}

/**
    是否包含另外一个字符串*/
func (this *Str)Strpos(Compare_String string)bool{

}

/**
    返回指定索引的那个字符,从0开始数起*/
func (this *Str)One_Str_by_Index(Index int)string{

}

/**
    变成小写*/
func (this *Str)Lowercase()string{

}
func (this *Str)Lowercase_this() *Str{
    this.Lowercase();
    return this;
}

